pkg/pool-weighted/contracts/managed/ManagedPool.sol

```diff
@@ -158,6 +158,17 @@ contract ManagedPool is ManagedPoolSettings {
         // solhint-enable no-empty-blocks
     }
 
+    // We need to check the breakers for all tokens on joins and exits (including join and exit swaps), since any
+    // change to the virtualSupply affects all BPT prices. For a multi-token join or exit, we will have a set of
+    // balances and delta amounts. For a join/exitSwap, only one token balance is changing. We can use the same
+    // data for both: in the single token swap case, the other token `amounts` will be zero.
+    struct PoolDelta {
+        IERC20[] tokens;
+        uint256[] normalizedWeights;
+        uint256[] balances;
+        uint256[] amounts;
```

Comment:
```suggestion
        uint256[] finalBalances;
```
Not required but we may want to mutate the balances into the post join/exit balances and just pass that.